cbot_actuators/src/sendThrusterData.cpp

```diff
@@ -0,0 +1,68 @@
+/************************* Thruster node ***************************
+
+-> Takes common mode and differential modes as inputs, peforms 
+  desired conversion and send commands to thrusters
+
+*******************************************************************/
+
+
+#include "ros/ros.h"
+#include "cbot_common/serial.hpp"
+#include "cbot_ros_msgs/ThrusterData.h"
+
+int HIL_ON;
+struct termios oldtio, newtio;
+int fd_hil;
+
+float T1,T2,T3,T4;
+
+void thrusterCallback(const cbot_ros_msgs::ThrusterData::ConstPtr& msg)
+{
+    T1 = msg->T1;
+    T2 = msg->T2;
+    T3 = msg->T3;
+    T4 = msg->T4;
+}
+
+void timerCallbck(const ros::TimerEvent& event){
+    ros::param::get("/HIL_ON",HIL_ON);
```

Comment:
ROS params should not be read in callbacks. Better approach is to have a service which sets the desired variable and callback just reads that variable instead of rosparam.